src/object/object_domain.c

```diff
@@ -9277,7 +9277,9 @@ tp_value_cast_internal (const DB_VALUE * src, DB_VALUE * dest, const TP_DOMAIN *
 	    {
 	      status = DOMAIN_INCOMPATIBLE;
 	    }
-	  else if (data_stat == DATA_STATUS_TRUNCATED && coercion_mode == TP_IMPLICIT_COERCION)
+	  else if (data_stat == DATA_STATUS_TRUNCATED &&
+              (prm_get_bool_value (PRM_ID_CHECK_TYPE_TRANS_STRICT) == true
+              || coercion_mode == TP_IMPLICIT_COERCION))
```

Comment:
In the "alter table" clause, a truncated string encounters this routine in case that casting function is applied. In that case, coercion mode is TP_IMPLICIT_COERCION and even though PRM_ID_CHECK_TYPE_TRANS_STRICT is fase it should be processed with coercion error.
https://github.com/CUBRID/cubrid-testcases/blob/741902a752cbd195cb44e97bb084fff2423c8c3b/sql/_30_banana_pie_qa/issue_20399_default_ext/cases/alter_set_add_misc.sql

_alter table t5 alter c1 set default to_char(systimestamp, 'cc yyyy yy q mm month mon dd day dy am pm a.m. p.m. hh hh12 hh24 mi ss');_

Reply:
I think it should also be modified to cast DB_TYPE_VARBIT into DB_TYPE_VARCHAR 
```
   case DB_TYPE_BIT:
    case DB_TYPE_VARBIT:
      switch (original_type)
	{
	case DB_TYPE_VARCHAR:
	case DB_TYPE_CHAR:
	case DB_TYPE_NCHAR:
	case DB_TYPE_VARNCHAR:
....
                  else if (data_stat == DATA_STATUS_TRUNCATED && coercion_mode == TP_IMPLICIT_COERCION)
		      {
			status = DOMAIN_OVERFLOW;
		      }
```

Reply:
good point.
I modified 2 points for similar case (default case: example, 0x1234 to bit varying(16))
the case you point out is run at cub_server (like as cast('12345' as bit varying(16))